packages/backend/src/studio-api-impl.ts

```diff
@@ -71,6 +72,8 @@ export class StudioApiImpl implements StudioAPI {
     const recipe = this.catalogManager.getRecipes().find(recipe => recipe.id === recipeId);
     if (!recipe) throw new Error('Not found');
 
+    await this.telemetryLogUsage('recipe.pull', { 'recipe.id': recipe.id, 'recipe.name': recipe.name });
```

Comment:
Not sure the telemetry event should be located here which is a gateway or we need to better handle errors